src/components/AmountInput.tsx

```diff
@@ -1,35 +1,65 @@
-import clsx from 'clsx';
+import { useState } from 'react';
+import { useAtomValue } from 'jotai';
+import { displayFunctionsAtom } from 'store/app';
+import { stringifyValue, parseAsValue } from '@agoric/ui-components';
+import { AssetKind } from '@agoric/ertp';
+import type { Brand } from '@agoric/ertp/src/types';
+import StyledInput from './StyledInput';
 
 type Props = {
+  value?: bigint | null;
   label?: string;
   error?: string;
+  brand?: Brand | null;
+  onChange: (value: bigint) => void;
   disabled?: boolean;
 };
 
-const AmountInput = ({ label, error, disabled = false }: Props) => {
+const AmountInput = ({
+  value,
+  onChange,
+  brand,
+  label,
+  error,
+  disabled = false,
+}: Props) => {
+  const { getDecimalPlaces } = useAtomValue(displayFunctionsAtom) ?? {};
+
+  const decimalPlaces =
+    (brand && getDecimalPlaces && getDecimalPlaces(brand)) ?? 0;
+
+  const amountString = stringifyValue(value, AssetKind.NAT, decimalPlaces);
+
+  const [fieldString, setFieldString] = useState(
+    value === null ? '' : amountString,
+  );
+
+  const handleInputChange: React.ChangeEventHandler<HTMLInputElement> = ev => {
+    const str = ev.target?.value?.replace('-', '').replace('e', '');
```

Comment:
I meant to ask in my last review… why remove '-' and 'e'? some examples of what these string values would be might help the reader

Reply:
Added a comment